flair/datasets/text_text.py

```diff
@@ -520,6 +520,476 @@ def tsv_from_eval_dataset(self, folder_path: Union[str, Path]):
+            base_path: Union[str, Path] = None,
+            max_tokens_per_doc=-1,
+            max_chars_per_doc=-1,
+            use_tokenizer=True,
+            in_memory: bool = True,
+            sample_missing_splits: bool = True
+    ):
+        """
+        Creates a DataPairCorpus for the Multi-Genre Natural Language Inference Corpus (MNLI)
+        from GLUE benchmark (https://gluebenchmark.com/tasks). Entailment annotations are: 
+        entailment, contradiction, neutral. This corpus includes two dev sets mathced/mismatched 
+        and two unlabeled test sets: eval_dataset_matched, eval_dataset_mismatched.
+        """
+
+        if type(base_path) == str:
+            base_path: Path = Path(base_path)
+
+        dataset_name = "glue"
+
+        # if no base_path provided take cache root
+        if not base_path:
+            base_path = flair.cache_root / "datasets"
+        data_folder = base_path / dataset_name
+
+        data_file = data_folder / "MNLI/train.tsv"
+
+        # if data is not downloaded yet, download it
+        if not data_file.is_file():
+            # get the zip file
+            zipped_data_path = cached_path(
+                "https://dl.fbaipublicfiles.com/glue/data/MNLI.zip",
+                Path("datasets") / dataset_name
+            )
+
+            unpack_file(
+                zipped_data_path,
+                data_folder,
+                mode="zip",
+                keep=False
+            )
+
+            # reorder dev datasets to have same columns as in train set: 8, 9, and 11
+            # dev sets include 5 different annotations but we will only keep the gold label
+            for dev_filename in ["dev_matched.tsv", "dev_mismatched.tsv"]:
+
+                temp_file = str("temp_" + dev_filename)
+                os.rename(str(data_folder / "MNLI" / dev_filename),
+                          str(data_folder / "MNLI" / temp_file))
+
+                with open(data_folder / "MNLI" / dev_filename, "a") as out_file, open(
+                    data_folder / "MNLI" / temp_file) as in_file:
+                    for line in in_file:
+                        fields = line.split('\t')
+                        reordered_columns = '\t'.join(fields[column_id] for column_id in range(11))
+                        reordered_columns += '\t' + fields[15]
+                        out_file.write(reordered_columns)
+                os.remove(str(data_folder / "MNLI" / temp_file))
+
+            # rename test file to eval_dataset, since it has no labels
+            os.rename(str(data_folder / "MNLI/test_matched.tsv"),
+                      str(data_folder / "MNLI/eval_dataset_matched.tsv"))
+            os.rename(str(data_folder / "MNLI/test_mismatched.tsv"),
+                      str(data_folder / "MNLI/eval_dataset_mismatched.tsv"))
+
+        matched_suffix = "matched" if evaluate_on_matched else "mismatched"
+        
+        dev_dataset = "MNLI/dev_" + matched_suffix + ".tsv"
```

Comment:
I think there is aproblem with the path here. I get an assertion error when first initializing the corpus. Maybe one 'MNLI' too much.

Reply:
Indeed! Thanks for pointing that out. I fixed the path for MNLI and trained the small electra model with it. MNLI seems to work now : )